src/Microsoft.Fast.Components.FluentUI/Components/NavMenu/FluentNavMenuGroup.razor.cs

```diff
@@ -108,38 +112,42 @@ public FluentNavMenuGroup()
         .AddStyle(Style)
         .Build();
 
-    private bool HasCollapsedIcon => !string.IsNullOrWhiteSpace(Icon) || CollapsedIconContent is not null;
-    private bool HasExpandedIcon => !string.IsNullOrWhiteSpace(Icon) || ExpandedIconContent is not null;
+    private bool HasIcon => !string.IsNullOrWhiteSpace(Icon) || IconContent is not null;
 
-    protected override void OnParametersSet()
-    {
-        NavMenu.AddNavMenuGroup(this);
+    internal bool HasNavMenuGutterIcon => !string.IsNullOrWhiteSpace(NavMenuGutterIcon) || NavMenuGutterIconContent is not null;
 
-        if (_expanded != Expanded)
-        {
-            _expanded = Expanded;
-            if (OnExpandedChanged.HasDelegate)
-                OnExpandedChanged.InvokeAsync(_expanded);
-        }
+    public override async Task SetParametersAsync(ParameterView parameters)
+    {
+        bool originalExpanded = Expanded;
+        await base.SetParametersAsync(parameters);
+        if (Expanded != originalExpanded)
+            await OnExpandedChanged.InvokeAsync(Expanded);
     }
```

Comment:
Override SetParametersAsync instead of OnParametersSet, so we can track before/after values without having to have additional private fields to store the previous state.